lib/picoruby/build.rb

```diff
@@ -79,6 +79,15 @@ def picoruby
       end
       cc.include_paths << gems['mruby-compiler2'].dir + "/lib/prism/include"
 
+      if /darwin/ =~ RUBY_PLATFORM && system("which brew > /dev/null 2>&1") # for macOS && homebrew
+        linker.library_paths << File.join(`brew --prefix`.chomp, "/lib")
+
+        openssl_path = `brew --prefix openssl`.chomp
+        if openssl_path != "" && openssl_path != nil
+          cc.include_paths << File.join(openssl_path, "include")
+        end
+      end
```

Comment:
@ginkouno Thank you very much for the patch!!

I don't know much about Mac, but I feel that it's not good to have code that only depends on Homebrew.
Is it possible to write it like this? (I don't have a Mac, so I haven't tested this code at all)

```ruby
if RUBY_PLATFORM.include?("darwin")
  openssl_prefix = ENV['OPENSSL_PREFIX']
  if openssl_prefix # User specific path
    cc.include_paths << File.join(openssl_prefix, "include")
    linker.library_paths << File.join(openssl_prefix, "lib")
  elsif system("which brew > /dev/null 2>&1")  # Homebrew
    linker.library_paths << File.join(`brew --prefix`.chomp, "/lib")
    openssl_path = `brew --prefix openssl`.chomp
    if openssl_path != "" && !openssl_path.nil?
      cc.include_paths << File.join(openssl_path, "include")
    end
  elsif system("which port > /dev/null 2>&1")  # MacPorts
    macports_prefix = "/opt/local"
    linker.library_paths << File.join(macports_prefix, "lib")
    openssl_path = File.join(macports_prefix, "include")
    if File.directory?(File.join(openssl_path, "openssl"))
      cc.include_paths << openssl_path
    end
  else  # Fallback
    ["/usr/local/opt/openssl/include", "/usr/local/include"].each do |path|
      if File.directory?(File.join(path, "openssl"))
        cc.include_paths << path
        lib_path = path.sub('/include', '/lib')
        linker.library_paths << lib_path if File.directory?(lib_path)
        break
      end
    end
  end
end
```

Reply:
In my experience (10 years of development on MacOS and development experience at around 5 companies), there were no workplaces that used anything other than homebrew, so I think that if you only support homebrew, you can cover a lot of ground for the time being.

However, there is definitely a risk, so I would like to respond as you have suggested.

However, there is a risk of breaking the development environment if you install libraries on a PC that has homebrew installed, so it would be helpful if you could narrow it down to specifying it in the environment variables and specifying it in homebrew. :pray:

After https://github.com/picoruby/picoruby/pull/181 is merged, I will implement it in the appropriate location.

Reply:
I'd suggest removing dependency on even Homebrew.
Now I feel macOS users should specify library paths through ENV

Reply:
> Now I feel macOS users should specify library paths through ENV

That's right!
I was able to successfully build on MacOS like this,

`LIBRARY_PATH=/opt/homebrew/lib/ C_INCLUDE_PATH=/opt/homebrew/include/ rake -v`

So I'm going to close this PR. Thank you for your help 🙏